debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java

```diff
@@ -148,7 +149,7 @@ private void takeSnapshot(Consumer<SourceRecord> consumer) {
             // we're locking in SHARE UPDATE EXCLUSIVE MODE to avoid concurrent schema changes while we're taking the snapshot
             // this does not prevent writes to the table, but prevents changes to the table's schema....
             schema.tables().forEach(tableId -> statements.append("LOCK TABLE ")
-                                                         .append(tableId.toString())
+                                                         .append(tableId.toQuotedId(PostgresTableIdTransformer.INSTANCE))
```

Comment:
More specifically, I'm wondering whether the id transformer shouldn't be stored within the given `TableId` instance. Then `toString()` could be used here, as before, and internally it'd make sure to whether add the quotes or not. Does that make sense?